packages/compass-e2e-tests/index.js

```diff
@@ -35,13 +35,17 @@ async function main() {
     mocha.addFile(path.join(__dirname, testPath));
   });
 
-  const failures = await promisify(mocha.run.bind(mocha))();
-
-  process.exitCode = failures ? 1 : 0;
+  try {
+    const failures = await promisify(mocha.run.bind(mocha))();
+    process.exitCode = failures ? 1 : 0;
+  } catch (err) {
+    console.error('mocha threw', err); // we never see failures. mocha throws 1
+    process.exitCode = 1;
+  }
```

Comment:
Ohhh, it's my mistake, mocha.run doesn't follow Node.js callback API pattern, you're right, we never get here. [`mocha.run`](https://mochajs.org/api/mocha#run) will call `fn` with numebr of failed tests and this line will throw

Reply:
I realised :) Was going to just discuss with you how best to change this. I could just drop the promisify and deal with the callback manually?

Reply:
Yeah, it's the last call in the method anyway, doing what they are doing in the docs should be reasonable enough here

Reply:
Done.